torch/csrc/jit/symbolic_script.cpp

```diff
@@ -1261,51 +1270,83 @@ const std::vector<std::string> functions = {
+                            use_scale_factor: Optional[bool]):
             def backward(grad_output):
                 if align_corners is None:
                     align_corners = False
-                grad_self = AD_interpolate_backward(grad_output, input, mode, align_corners)
-                return grad_self, None, None, None, None
+                if use_scale_factor is None:
+                    use_scale_factor = False
+                input_dim = len(input.size())
+                if scale_factor is None or not use_scale_factor:
+                    scales = [-1.0 for i in range(input_dim-2)]
+                else:
+                    scales = [scale_factor[i] for i in range(input_dim-2)]
+                grad_self = AD_interpolate_backward(grad_output, input, mode, align_corners, scales)
+                return grad_self, None, None, None, None, None
 
             return torch.__interpolate(input, size, scale_factor, mode, align_corners), backward
 
         def __interpolate_1(input,
                             size: Optional[List[int]],
                             scale_factor: Optional[List[float]],
                             mode: str='nearest',
-                            align_corners: Optional[bool]):
+                            align_corners: Optional[bool],
+                            use_scale_factor: Optional[bool]):
             def backward(grad_output):
                 if align_corners is None:
                     align_corners = False
-                grad_self = AD_interpolate_backward(grad_output, input, mode, align_corners)
-                return grad_self, None, None, None, None
+                if use_scale_factor is None:
+                    use_scale_factor = False
+                input_dim = len(input.size())
+                if scale_factor is None or not use_scale_factor:
+                    scales = [-1.0 for i in range(input_dim-2)]
+                else:
+                    scales = [scale_factor[i] for i in range(input_dim-2)]
+                grad_self = AD_interpolate_backward(grad_output, input, mode, align_corners, scales)
+                return grad_self, None, None, None, None, None
 
             return torch.__interpolate(input, size, scale_factor, mode, align_corners), backward
 
         def __interpolate_2(input,
                             size: Optional[int],
                             scale_factor: Optional[float],
                             mode: str='nearest',
-                            align_corners: Optional[bool]):
+                            align_corners: Optional[bool],
+                            use_scale_factor: Optional[bool]):
             def backward(grad_output):
                 if align_corners is None:
                     align_corners = False
-                grad_self = AD_interpolate_backward(grad_output, input, mode, align_corners)
-                return grad_self, None, None, None, None
+                if use_scale_factor is None:
```

Comment:
Copy-paste across the four different `interpolate`  would be better factored into a helper function. Helper functions are at the top of this file.